### PR TITLE
Improved API documentation

### PR DESCRIPTION
**Pull request by Fir0w**

### CODE REVIEW

File: server/controllers/voteController.js [hunk 1]
```diff
@@ -2,6 +2,12 @@ import Post from '../models/postModel.js';
 import Vote from '../models/voteModel.js';
 
 
+// @desc This API Creates a Vote and updates the Vote in DB
+// route POST /api/posts/vote/
+// @accesss Private
+// @returns {object} 200 - Returns an object that returns a message
```
Comment (VicDosya) [inline server/controllers/voteController.js:8]: Youre missing more status 200 objects (upvote, downvote) must be accurate.

File: server/controllers/postController.js [hunk 2]
```diff
@@ -22,17 +28,28 @@ const createPost = async (req, res) => {
...
 
+// @desc This API Deletes a Post from DB
+// route GET /api/posts/
+// @accesss Private
+// @returns {object} 200 - Returns an object that returns a message 
+// @returns {object} 500 - Returns an object that returns a message "something went wrong"
+// @returns {object} 500 - Returns an object that returns a message "Internal Server Error"
 const deletePost = async (req, res) => {
 
     if (req.query.userId !== req.user._id.toString())
```
[the remark this answers is not in the document]
Comment (VicDosya) [reply server/controllers/postController.js:55]: are you checking if user id in the url query is the same as the user in his auth ?
this is all a client-side check ?

[the remark this answers is not in the document]
Comment (Fir0w) [reply server/controllers/postController.js:55]: I'm checking if a different user than the user that created the post is trying to delete it.

[the remark this answers is not in the document]
Comment (VicDosya) [reply server/controllers/postController.js:55]: yes but both of the user data coming from the client side (req.query.userId and req.user._id).
There is no server side check (in the User model or Post model), am I wrong ?

[the remark this answers is not in the document]
Comment (Fir0w) [reply server/controllers/postController.js:55]: I mean the req.user._id comes from the cookie that is encrypted

**Review by VicDosya — APPROVED** [review]
😀

**Pull request merged**